### PR TITLE
fix: payment reconciliation get negative outstanding invoices.

### PR DESCRIPTION
**Pull request by dj12djdjs**
### Steps to Reproduce
1. Create Sales Invoice
2. Create Payment Entry Against Sales Invoice
3. Create Credit Note Against Sales Invoice (Sales Invoice Outstanding Amount is now negative.
4. Go to Payment Reconciliation form the negative sales invoice should be assignable as payment.

### CODE REVIEW

**Comment by barredterra** [comment]
@dj12djdjs can you summarize he current status, which PRs can be merged in which order, any outstanding issues, etc?

**Comment by aakvatech** [comment]
Are we  expecting this to be pushed forward?

**Comment by dj12djdjs** [comment]
Closing as it will be resolved by https://github.com/frappe/erpnext/pull/35330#issuecomment-1680465569
